fastapi/routing.py

```diff
@@ -168,7 +168,10 @@ def get_request_handler(
     is_coroutine = asyncio.iscoroutinefunction(dependant.call)
     is_body_form = body_field and isinstance(body_field.field_info, params.Form)
     if isinstance(response_class, DefaultPlaceholder):
-        actual_response_class: Type[Response] = response_class.value
+        if status_code == HTTP_204_NO_CONTENT:
+            actual_response_class: Type[Response] = Response
+        else:
+            actual_response_class = response_class.value
```

Comment:
the type annotation was dropped here

Reply:
Pylance (Pyright) called this a redefinition, I'll check if mypy throws the same errors and if not I'll add type annotation to the else clasue

Reply:
I found some solution, not best either but maybe better?